src/__tests__/FieldArray.spec.js

```diff
@@ -1182,6 +1186,58 @@ const describeFieldArray = (name, structure, combineReducers, expect) => {
         .toBe('No dogs')
     })
 
+    it('should rerender when depending value has updated', () => {
+      const store = makeStore({
+        testForm: {
+          values: {
+            dogs: [ {
+              name: 'Fido',
+              hasCollar: false
+            }, {
+              name: 'Snoopy',
+              hasCollar: false
+            } ]
+          }
+        }
+      })
+      const renderFieldArray =
+        createSpy(({ fields }) => (<div>
+          {fields.map((field, index) =>
+            <div key={index}>
+              {getIn(fields.get(index), 'hasCollar') && <span className="collar"></span>}
+              <Field name={`${field}.hasCollar`} component="input" type="checkbox" />
+              <Field name={`${field}.name`} component="input" />
+            </div>)}
+          <button className="add" onClick={() => fields.push()}>Add Dog</button>
+          <button className="remove" onClick={() => fields.pop()}>Remove Dog</button>
+        </div>)).andCallThrough()
+      class Form extends Component {
+        render() {
+          return <FieldArray name="dogs" component={renderFieldArray}/>
+        }
+      }
+      const TestForm = reduxForm({
+        pure: false,
+        form: 'testForm'
+      })(Form)
+      const dom = TestUtils.renderIntoDocument(
+        <Provider store={store}>
+          <TestForm/>
+        </Provider>
+      )
+
+      const checkbox = TestUtils.scryRenderedDOMComponentsWithTag(dom, 'input')
```

Comment:
Now, I show the `span` element, when checkbox is checked.

The checkbox field itself at the line 1208 is always visible.